src/main/java/edu/kis/powp/jobs2d/features/RecordingFeature.java

```diff
@@ -0,0 +1,76 @@
+package edu.kis.powp.jobs2d.features;
+
+import edu.kis.powp.appbase.Application;
+import edu.kis.powp.jobs2d.drivers.decorator.RecordingDriverDecorator;
+import edu.kis.powp.jobs2d.events.SelectRecordMacroMenuOptionListener;
+
+import javax.swing.*;
+import java.awt.event.ActionListener;
+import java.util.Arrays;
+
+public class RecordingFeature {
+	private static boolean isRecording = false;
+	private static final RecordingDriverDecorator driver =
+			new RecordingDriverDecorator(DriverFeature.getDriverManager().getCurrentDriver());
+
+	private static JCheckBoxMenuItem checkbox;
+
+	public static final String RECORD_MACRO = "Record Macro";
+	public static final String RECORDING = "Recording";
+	public static final String CLEAR_MACRO = "Clear Macro";
+	public static final String RUN_MACRO = "Run Macro";
+
+	private RecordingFeature() {}
+
+	public static void setupRecordingPlugin(Application app) {
+		ActionListener listener = new SelectRecordMacroMenuOptionListener();
+
+		app.addComponentMenu(RecordingFeature.class, RECORD_MACRO);
+		app.addComponentMenuElementWithCheckBox(RecordingFeature.class, RECORDING, listener, isRecording);
+		app.addComponentMenuElement(RecordingFeature.class, CLEAR_MACRO, listener);
+		app.addComponentMenuElement(RecordingFeature.class, RUN_MACRO, listener);
+
+		DriverFeature.getDriverManager().getChangePublisher().addSubscriber(driver);
+		checkbox = Arrays.stream(app.getFreePanel().getRootPane().getJMenuBar().getSubElements())
+				.flatMap(x -> Arrays.stream(x.getSubElements()))
+				.flatMap(x -> Arrays.stream(x.getSubElements()))
+				.filter(x -> x instanceof JCheckBoxMenuItem)
+				.map(x -> (JCheckBoxMenuItem) x)
+				.filter(x -> x.getActionCommand().equals(RECORDING))
+				.findFirst()
+				.orElseGet(JCheckBoxMenuItem::new);
+	}
+
+	public static void register() {
```

Comment:
Why is it public? 
The name is a little bit confusing.

Reply:
Changed to private and renamed